haskell/test/Lorentz/Contracts/Test/FA2.hs

```diff
@@ -0,0 +1,542 @@
+-- | Tests for FA2 interface.
+-- https://gitlab.com/tzip/tzip/-/blob/131b46dd89675bf030489ded9b0b3f5834b70eb6/proposals/tzip-12/tzip-12.md
+
+module Lorentz.Contracts.Test.FA2
+  ( OriginationParams (..)
+  , TokenDomain (..)
+  , defaultPermissionDescriptor
+  , fa2Spec
+  ) where
+
+import Data.Kind
+import qualified Data.Map as Map
+import Test.Hspec (Spec, describe, it)
+
+import Lorentz (mkView)
+import Lorentz.Contracts.Spec.FA2Interface as FA2
+import Lorentz.Test
+import Lorentz.Value
+import Util.Named
+
+wallet1, wallet2, wallet3, wallet4, wallet5, commonOperator :: Address
+wallet1 = genesisAddress1
+wallet2 = genesisAddress2
+wallet3 = genesisAddress3
+wallet4 = genesisAddress4
+wallet5 = genesisAddress5
+commonOperator = genesisAddress6
+
+data TokenDomain = SingleToken | MultiToken
```

Comment:
As I suggested in the respestive morley MR, let's remove it for now and just assume `SingleToken` everywhere.

Reply:
Have removed.